transaction_integration_test.go

```diff
@@ -707,3 +714,27 @@ func TestTransactionCreateWithCustomFields(t *testing.T) {
 		t.Fatalf("Returned custom fields doesn't match input, got %q, want %q", x, customFields)
 	}
 }
+
+func TestTransactionTaxExempt(t *testing.T) {
+	t.Parallel()
+
+	amount := NewDecimal(10000, 2)
+	txn, err := testGateway.Transaction().Create(&TransactionRequest{
+		Type:               "sale",
+		Amount:             amount,
+		TaxExempt:          true,
+		PaymentMethodNonce: FakeNonceTransactable,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	txn, err = testGateway.Transaction().Find(txn.Id)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !txn.TaxExempt {
+		t.Fatalf("Transaction did not return tax exempt")
+	}
+}
```

Comment:
I just ran the `TestTransactionCreateSubmitForSettlementAndVoid` test with the logger turned on that logs the XML sent and received, and this is what came back for the tax amount field:

```xml
...
<tax-amount nil="true"/>
<tax-exempt type="boolean">false</tax-exempt>
...
```

The XML field is there, but it has an attribute `nil="true"`.

This is probably a bug that the packages `Decimal` type (and other types) don't translate the field with `nil="true"` into a nil. Looking at some of the official SDKs they seem to translate the attribute to mean the field is nil. e.g. https://github.com/braintree/braintree_ruby/blob/master/lib/braintree/xml/parser.rb#L58-L59

I'd add the expectation here it's nil, and separately we can take a look at the `Decimal` type and see if it's safe to change it's behavior on `nil="true"`, and if so, fix that ahead of merging this PR. My motivation for suggesting this is if we merge it as it is and fix this later others may develop logic depending on getting back a zero Decimal rather than a nil Decimal.